HotelApi/Controllers/ReservationController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Http;
+using System.Web.Http.Results;
+using HotelBooking;
+using HotelBooking.BookingExternalService;
+using HotelBooking.HotelFactory;
+using HotelBooking.HotelManager;
+using HotelBooking.Logger;
+using HotelBooking.PaymentExternalService;
+
+namespace HotelApi.Controllers
+{
+    public class ReservationController : ApiController
+    {
+        private readonly HotelManager _manager;
+
+        public ReservationController()
+        {
+            IHotelFactory factory = new HotelFactory();
```

Comment:
Nice to have:

Could you please try to replace creating dependency by hand by one of Dependency Injection framework e.g. Niject?

Reply:
applied (after struggling for a couple of hours with nuget packages)